### ISSUE TITLE
When I type "un" into the flow launcher I get this exception

### ISSUE THREAD

**Issue by Dax0s** [falldeaf/unity-flowlauncher#1]
Flow Launcher version: 1.9.3
OS Version: Microsoft Windows NT 10.0.19043.0
IntPtr Length: 8
x64: True

Python Path:
Date: 07/12/2022 13:05:13
Exception:
System.Text.Json.JsonReaderException: 'G' is an invalid start of a value. LineNumber: 0 | BytePositionInLine: 0.
at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan1 bytes) at System.Text.Json.Utf8JsonReader.ConsumeValue(Byte marker) at System.Text.Json.Utf8JsonReader.ReadFirstToken(Byte first) at System.Text.Json.Utf8JsonReader.ReadSingleSegment() at System.Text.Json.Utf8JsonReader.Read() at System.Text.Json.JsonDocument.Parse(ReadOnlySpan1 utf8JsonSpan, JsonReaderOptions readerOptions, MetadataDb& database, StackRowStack& stack)
at System.Text.Json.JsonDocument.Parse(ReadOnlyMemory1 utf8Json, JsonReaderOptions readerOptions, Byte[] extraRentedBytes) at System.Text.Json.JsonDocument.Parse(ReadOnlyMemory1 json, JsonDocumentOptions options)
at System.Text.Json.JsonDocument.Parse(String json, JsonDocumentOptions options)
at Flow.Launcher.Plugin.UnityHelper.UnityHelper.Query(Query query)
at Flow.Launcher.Plugin.IPlugin.<>c__DisplayClass3_0.<Flow.Launcher.Plugin.IAsyncPlugin.QueryAsync>b__0() in /_/Flow.Launcher.Plugin/Interfaces/IPlugin.cs:line 35
at System.Threading.Tasks.Task1.InnerInvoke() at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) --- End of stack trace from previous location --- at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) --- End of stack trace from previous location --- at Flow.Launcher.Core.Plugin.PluginManager.<>c__DisplayClass22_0.<<QueryForPluginAsync>b__0>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 192 --- End of stack trace from previous location --- at Flow.Launcher.Infrastructure.Stopwatch.DebugAsync(String message, Func1 action) in C:\projects\flow-launcher\Flow.Launcher.Infrastructure\Stopwatch.cs:line 40
at Flow.Launcher.Core.Plugin.PluginManager.QueryForPluginAsync(PluginPair pair, Query query, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 211
at Flow.Launcher.ViewModel.MainViewModel.<>c__DisplayClass137_0.<g__QueryTask|2>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 623
--- End of stack trace from previous location ---
at Flow.Launcher.ViewModel.MainViewModel.QueryResults() in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 587
at System.Threading.Tasks.Task.<>c.b__140_0(Object state)
at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

Edit: Log file from today [2022-07-13.txt](https://github.com/falldeaf/unity-flowlauncher/files/9101060/2022-07-13.txt)

**Comment by falldeaf** [comment]
Can you run the command (in powershell):
`Get-UnityProjectInstance -BasePath [path/to/all/your/Unity projects] -Recurse | ConvertTo-Json`

And paste the output?

I also found a bug yesterday that if a project uses the option 'mixed' for asset serialization, the powershell script will complain and gunk up the json output. I need to handle json parsing errors better anyway. I'm just assuming I'll get good input but that's a terrible assumption, especially since this plugin requires an external powershell module.

**Comment by Dax0s** [comment]
Here are the warnings and exceptions:

WARNING: Could not read D:\Repos\Unity\Learning\Random\2D movement\ProjectSettings\ProjectSettings.asset, in the Unity
project try setting Editor Settings > Asset Serialiazation Mode to 'Force Text'.
An Exception was caught!
Exception Type: System.Management.Automation.RuntimeException
Exception Message: ProjectSettings is missing productName
WARNING: Could not read D:\Repos\Unity\Learning\Unity 2D course\Beginning\ProjectSettings\ProjectSettings.asset, in the
 Unity project try setting Editor Settings > Asset Serialiazation Mode to 'Force Text'.
An Exception was caught!
Exception Type: System.Management.Automation.RuntimeException
Exception Message: ProjectSettings is missing productName
WARNING: Could not read D:\Repos\Unity\Learning\Unity 2D course\Text101\ProjectSettings\ProjectSettings.asset, in the
Unity project try setting Editor Settings > Asset Serialiazation Mode to 'Force Text'.
An Exception was caught!
Exception Type: System.Management.Automation.RuntimeException
Exception Message: ProjectSettings is missing productName

And here's the command output: [output.txt](https://github.com/falldeaf/unity-flowlauncher/files/9104905/output.txt)

**Comment by falldeaf** [comment]
Ok yeah, 10-4, thanks for getting the output and posting. If you open those three projects and change the asset serialization type to 'force text' that will get the plugin working, if you want a quick-fix. But obviously this is still a bug that needs to be fixed the right way. I'll update the code sometime before this Friday to account for these issues.

**Comment by falldeaf** [comment]
I just pushed a fix for this to the repository and a release as well so it should automatically load the new one in FlowLauncher. The one caveat is, I'm using `$warningPreference = 'SilentlyContinue';` to suppress the error, but the problem will still be there. So if you have any Unity projects with asset serialization mode set to binary or mixed, the project name will be blank (Everything else will work fine. ) If you want to fix that as well, set the Asset serialization mode to 'force text'.

Sorry for the delay, I got covid and have been sick the past week. :/
